SNIP-25.md

```diff
@@ -0,0 +1,87 @@
+# SNIP-25 - Improved privacy for [SNIP-20](/SNIP-20.md) tokens
+
+This document outlines updates to the SNIP-20 specification with the introduction of `decoys`, which are designed to strengthen privacy measures by ensuring that account information remains concealed during transaction processes. This is accomplished by intermixing real accounts with 'decoy' accounts, thus making it difficult for potential attackers to discern the actual accounts being updated. Moreover, randomness is integrated into the execution messages, which offers an additional layer of protection by creating uncertainty in storage access patterns.
+
+Contracts that support the existing SNIP-20 standard are still considered
+compliant, and clients SHOULD be written so as to benefit from this feature
+when available.
+
+- [Rationale](#rationale)
+- [Messages](#messages)
+- [Batch Operations](#batch-operations)
+- [Randomness Appendix](#randomness-appendix)
+
+## Rationale
+
+Secret Network smart contracts provide computational privacy through the use of Trusted Execution Environments, which process encrypted data securely without revealing it publicly on chain. However, the patterns at which data is accessed when processing confidential transactions can leak crucial information. In the example of a SNIP-20, each account balance may be stored privately, but the access pattern is public, so an untrusted host OS can potentially learn private information by monitoring accesses to the key-value storage. For example, each user's balance is kept in a key-value store, and the encrypted key for each account stays the same. By keeping track of the keys fetched from the storage, one can figure out the receiver of a SNIP-20 transfer.
+
+![Storage Access Pattern](https://i.ibb.co/N1JsQrC/storage-access-pattern.png)
+
+In order to effectively incorporate decoys, it is necessary to randomly interweave decoy addresses with the real addresses, ensuring that the real addresses are included within the sequence. Decoys SHOULD be used for every message exposed to storage access pattern attacks, and for every transaction that does, it MUST include the `decoys` parameter. Provided decoy addresses MUST be treated and utilized as if they were genuine addresses and MUST be used in a way that is indistinguishable from real addresses. Execution messages which accept the `decoys` field MAY also accept the `entropy` field, which is an optional Binary used to introduce randomness by providing additional read and writes to the decoy addresses that are included in the message. How `entropy` is implemented is left to the developer's discretion. For example, one MAY include the `decoys` field and not the `entropy` field and use Secret VRF for randomness.
+
+## Messages
+
+The existing `Redeem`, `Deposit`, `Transfer`, `Send`, `Burn`, `TransferFrom`, `SendFrom`, `BurnFrom`, and `Mint` messages now SHOULD accept the `decoys` field.
+
+Example `Transfer` message, with optional reference to `decoys` included.
```

Comment:
It may be better to specify what makes it optional. Something like: "Example Transfer message, where the decoys field is optional."